### PR TITLE
Fix add fragment

### PR DESCRIPTION
**Pull request by JustAnOtherCoder13**
add fragment work well when add or update.
test for new update methods done and passed

missing : 

force user to fill edit text
add loader when add button click
get date when enter on market and persist.
resolve bug on rotation

### CODE REVIEW

**Comment by JustAnOtherCoder13** [comment]
got trouble on my edit text in custom view else everything work fine

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/main/BaseFragment.java [hunk 1]
```diff
@@ -44,6 +50,15 @@ protected void setUpdateButtonIcon(boolean isForUpdate) {
         mainActivity.initUpdateButton(isForUpdate);
     }
 
+    protected void hideSoftKeyboard(View view) {
+        mainActivity.hideSoftKeyboard(view);
+    }
+
+    protected void playLoader(boolean isVisible) {
```
Comment (fodeguirassy) [inline app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/main/BaseFragment.java:56]: It seems that you don't need mLoader

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/utils/customView/AddPropertyInformationCustomView.java [hunk 2]
```diff
@@ -43,13 +44,21 @@ private void initView(TypedArray attributes) {
     }
 
 
-    public static void setValueText( View view,String text) {
```
Comment (fodeguirassy) [inline app/src/main/java/com/openclassrooms/realestatemanager/presentation/utils/customView/AddPropertyInformationCustomView.java:46]: Method does not need to be static
editText is a global variable inside the customView

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/utils/customView/AddPropertyInformationCustomView.java [hunk 3]
```diff
@@ -43,13 +44,21 @@ private void initView(TypedArray attributes) {
...
+    public static void setValueText(@NonNull View view, String text) {
         EditText editText = view.findViewById(R.id.add_property_information_custom_view_value);
         editText.setText(text);
     }
 
-    public static String getValueForView(View view){
+
+    @NonNull
```
Comment (fodeguirassy) [inline app/src/main/java/com/openclassrooms/realestatemanager/presentation/utils/customView/AddPropertyInformationCustomView.java:50]: Method does not need to be static
editText should be global to the CustomView. And you won't need to pass the view as parameter

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/utils/customView/AddPropertyInformationCustomView.java [hunk 4]
```diff
@@ -43,13 +44,21 @@ private void initView(TypedArray attributes) {
...
+    public static String getValueForView(@NonNull View view) {
         EditText editText = view.findViewById(R.id.add_property_information_custom_view_value);
         return editText.getText().toString();
     }
+
+    @NonNull
```
Comment (fodeguirassy) [inline app/src/main/java/com/openclassrooms/realestatemanager/presentation/utils/customView/AddPropertyInformationCustomView.java:55]: Same here

**Review by fodeguirassy — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/main/BaseFragment.java (hunk 1 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/main/BaseFragment.java:56]: you're right. done

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/utils/customView/AddPropertyInformationCustomView.java (hunk 3 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/openclassrooms/realestatemanager/presentation/utils/customView/AddPropertyInformationCustomView.java:50]: done

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/utils/customView/AddPropertyInformationCustomView.java (hunk 4 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/openclassrooms/realestatemanager/presentation/utils/customView/AddPropertyInformationCustomView.java:55]: done

**Comment by JustAnOtherCoder13** [comment]
change done but same bug on rotate persist.
log in setView show that the good text is passed to the method.

i'll go on add media branche, work arround this bug later.

**Pull request merged**